ocaml/xapi/pool_features_helpers.ml

```diff
@@ -58,17 +58,19 @@ let rec compute_additional_restrictions all_host_params = function
       []
   | flag :: rest ->
       let switches =
-        List.map
+        List.exists
           (function
-            | params ->
-                if List.mem_assoc flag params then
-                  bool_of_string (List.assoc flag params)
-                else
+            | params -> (
+              match List.assoc_opt flag params with
+              | Some x ->
+                  bool_of_string x
+              | None ->
                   true
```

Comment:
bool_of_string raises, I think I'd prefer if it fell back to the value not existing when this happens:
```suggestion
               List.assoc_opt flag params
               |> Fun.flip Option.bind bool_of_string_opt
               |> Option.value ~default:true
```